src/models/step_arguments/data_table.js

```diff
@@ -1,5 +1,31 @@
 import _ from 'lodash'
 
+const truths = ['true', 'TRUE', 'True', 'yes', 'YES', 'Yes', 'y', 'Y', '1']
+const trueFalse = (v) => !!~truths.indexOf(v)
+const asIs = (v) => v
+const list = (v) => v.split(',')
+const date = (v) => new Date(v)
+
+const types = {
+  'string':  asIs,
+  'str':     asIs,
+  '':        asIs,
+  'int':     parseInt,
+  'integer': parseInt,
+  'double':  parseInt,
+  'float':   parseFloat,
+  'number':  parseFloat,
+  'bool':    trueFalse,
+  'boolean': trueFalse,
+  'y/n':     trueFalse,
+  'bit':     trueFalse,
+  date,
+  'datetime':date,
+  'array':   list,
+  list,
+  'json':    JSON.parse
```

Comment:
I'd prefer we get rid of all the duplicates here. Thus only have 'string', 'number' (encapsulates int / float), 'boolean', 'date', 'list', (I don't like array since we are limited to an array of strings) and 'json'.